stack_data/core.py

```diff
@@ -669,6 +673,23 @@ def lines(self) -> List[Union[Line, LineGap]]:
         if not pieces:
             return []
 
+        # In case the node of interest spans multiple lines,
+        # we do a first pass through to determine what should
+        # be the common indentation of the markers
+        indents = []
+        for i, piece in enumerate(pieces):
+            lines = [
+                Line(self, i)
+                for i in piece
+            ]  # type: List[Line]
+            for line in lines:
+                for line_range in line.executing_node_ranges:
+                    if line.is_current:
+                        indents.append(line_range.start)
```

Comment:
This doesn't look right, rather than `is_current` I think it should be about whether it's the first line in the executing node, i.e. the only line that might have non-whitespace before the executing node starts.

Reply:
If you look at the second example above, the alignment is done with `x[0` which, I think looks better than if it were with `x=...` on the previous line.

Reply:
The idea I had was to minimize the amount of white space being highlighted, while keeping a vertical alignment of all the highlighted code.

Reply:
Yes, but `x[0` is the start of the executing node in that case so the example wouldn't change. Basically your examples always have the current line being the same as the first highlighted line.

Reply:
Isn't this always the case?

Reply:
```python
import stack_data
stack_data.Formatter().set_hook()

lst = []
print(lst
      .pop())
```

In 3.9:

```
       1 | import stack_data
       2 | stack_data.Formatter().set_hook()
       4 | lst = []
-->    5 | print(lst
                 ^^^
       6 |       .pop())
           ^^^^^^^^^^^^
```

But in 3.10:

```
       1 | import stack_data
       2 | stack_data.Formatter().set_hook()
       4 | lst = []
       5 | print(lst
                 ^^^
-->    6 |       .pop())
           ^^^^^^^^^^^^
```

https://bugs.python.org/issue39316

Reply:
Ok, I was able to modify it so that I get the following for Python 3.10.
```
       1 | import stack_data
       2 | stack_data.Formatter().set_hook()
       4 | lst= []
       5 | print(lst
                 ^^^
-->    6 |       .pop())
                 ^^^^^^
```
and this for 3.9
```
       1 | import stack_data
       2 | stack_data.Formatter().set_hook()
       4 | lst= []
-->    5 | print(lst
                 ^^^
       6 |       .pop())
                 ^^^^^^
I```
A couple more examples for good measure:
```
-->    5 | print(lst
                 ^^^
       6 |          .pop())
                 ^^^^^^^^^
```
```
-->    5 | print(lst
                 ^^^
       6 |    .pop())
              ^^^^^^
```
Other than the indicator for the current line, I get the same markers for the indentation for both 3.9 and 3.10.

I will need to do more testing.

Unrelated: I've never attempted to do a multi-commit PR  (and I didn't create a separate branch). If I do another commit in my fork followed by a PR, will it be smart enough to add it to this one?
(I won't be able to do significantly more work on this right now.)

Reply:
A final one for tonight:
```
       4 | print([1, 2,
                 ^^^^^^
       5 |     3, 4, 5]
               ^^^^^^^^
-->    6 |          .pop(8))
               ^^^^^^^^^^^^
```

Reply:
It should be all ready now. 

Other than refactoring, I had to make an additional change so that using it to use markers for frames directly (as opposed as arising from a traceback), like you did in some tests with `<var>...</var>`, was broken by my code. Everything should now be fixed.